compiler/rustc_type_ir/src/lib.rs

```diff
@@ -90,8 +94,9 @@ rustc_index::newtype_index! {
     /// is the outer fn.
     ///
     /// [dbi]: https://en.wikipedia.org/wiki/De_Bruijn_index
-    #[derive(HashStable_Generic)]
+    #[cfg_attr(feature = "nightly", derive(HashStable_Generic))]
     #[debug_format = "DebruijnIndex({})"]
+    #[gate_rustc_only]
```

Comment:
Wait, am I confused, or does this mean that `DebruijnIndex` (and `UniverseIndex`) below are nightly (rustc) only?

Reply:
If so, why? If there is no reason, this shouldn't be gated.

If not, ignore me :)

Reply:
No, sorry. The attribute just means "emit all the nightly stuff (e.g. `Step` and `Encodable`/`Decodable`) with a `cfg(feature = "nightly")` so it can be compiled on stable too". Maybe it should be renamed, but can be later.